src/Core/AdminConsole/Services/Implementations/OrganizationService.cs

```diff
@@ -728,6 +731,32 @@ private async Task ValidateSignUpPoliciesAsync(Guid ownerId)
         }
     }
 
+    public async Task InitiateDeleteAsync(Organization organization, string orgAdminEmail)
+    {
+        var orgAdmin = await _userRepository.GetByEmailAsync(orgAdminEmail);
+        if (orgAdmin == null)
+        {
+            throw new BadRequestException("Org admin not found.");
+        }
+        var orgAdminOrgUser = await _organizationUserRepository.GetDetailsByUserAsync(orgAdmin.Id, organization.Id);
+        if (orgAdminOrgUser == null || orgAdminOrgUser.Status != OrganizationUserStatusType.Confirmed ||
+            (orgAdminOrgUser.Type != OrganizationUserType.Admin && orgAdminOrgUser.Type != OrganizationUserType.Owner))
+        {
+            throw new BadRequestException("Org admin not found.");
+        }
+        var token = _orgDeleteTokenDataFactory.Protect(new OrgDeleteTokenable(organization, 1));
+        await _mailService.SendInitiateDeleteOrganzationEmailAsync(orgAdminEmail, organization, token);
+    }
+
+    public async Task DeleteAsync(Organization organization, string token)
+    {
+        if (!_orgDeleteTokenDataFactory.TryUnprotect(token, out var data) && data.IsValid(organization))
```

Comment:
This makes more sense to me, right?
```suggestion
        if (!_orgDeleteTokenDataFactory.TryUnprotect(token, out var data) || !data.IsValid(organization))
```